ring/replication_set_tracker.go

```diff
@@ -157,9 +168,14 @@ func (t *defaultResultTracker) startMinimumRequests() {
 }
 
 func (t *defaultResultTracker) startAdditionalRequests() {
+	t.startAdditionalRequestsDueTo("hedging")
+}
+
+func (t *defaultResultTracker) startAdditionalRequestsDueTo(reason string) {
 	if len(t.pendingInstances) > 0 {
 		// There are some outstanding requests we could make before we reach maxErrors. Release the next one.
 		i := t.pendingInstances[0]
+		level.Debug(t.logger).Log("msg", "starting request to instance due to "+reason, "instance", i.Addr)
```

Comment:
[nit] Have you considered like this?
```suggestion
		level.Debug(t.logger).Log("msg", "starting request to instance", "reason", reason, "instance", i.Addr)
```